### PR TITLE
chore(repo): deprecate testID and add testId - 2

### LINKED ISSUE #2458: [Repo] Add `data-testid` attribute and `testId` props - 2
We need to add `data-testid` attribute and `testId` props to all component's major sections as well as write test to ensure these selectors do not break when code is updated. Follow discussion outlined in [this](https://github.com/carbon-design-system/carbon-addons-iot-react/issues/1001) epic. Skip if it is a carbon component that we do not extend.

- [x] Checkbox
- [x] CodeSnippet
- [x] ColorDropdown
- [x] ComboBox
- [x] ComboChartCard
- [x] ComposedModal
- [x] ComposedStructuredList
- [x] ContentSwitcher

### PR DESCRIPTION
**Pull request by kevinsperrine**
Closes #2458 

Depends on #2457

**Summary**

Adds testId prop to element that don't have it, and marks the existing testID prop as deprecated. It also uses the passed testID or testId to create usable test ids for nested components and make them easier to test.

**THIS PR SHOULD NOT CONTAIN ANY BREAKING CHANGES, SO THAT IT CAN BE ROLLED OUT BEFORE V3. IT NEEDS A GOOD EYE TO CONFIRM THAT WITH SO MANY CHANGES.**

Change List (commits, features, bugs, etc)

- Checkbox
- CodeSnippet
- ColorDropdown
- ComboBox
- ComboChartCard
- ComposedModal
- ComposedStructuredList
- ContentSwitcher

**Acceptance Test (how to verify the PR)**

- storyshots should now contain more data-testid attributes in their output
- testId should be used on all props for the components mentioned above and not testID
- testID should throw a deprecation warning

**Regression Test (how to make sure this PR doesn't break old functionality)**

- All tests still pass as expected.
- testID or testId should both be valid to maintain existing tests without breaking.

### CODE REVIEW

File: packages/react/src/components/ColorDropdown/ColorDropdown.jsx [hunk 1]
```diff
@@ -47,8 +48,14 @@ const propTypes = {
   translateWithId: PropTypes.func,
   /** The selected color, use to set initial color */
   selectedColor: colorPropType,
+  // TODO: remove deprecated testID in v3.
+  // eslint-disable-next-line react/require-default-props
+  testID: deprecate(
```
Comment (davidicus) [inline packages/react/src/components/ColorDropdown/ColorDropdown.jsx:53]: I believe this component is still experimental so we can probably swap out ID without deprecate.

**Review by davidicus — APPROVED** [review]
Seems good to go. If you want to wait to remove the deprecation at this point since its already there I am fine with that

**Comment by kevinsperrine** [comment]
> Seems good to go. If you want to wait to remove the deprecation at this point since its already there I am fine with that

Might as well just leave it now. We can take it out with everything else in V3.

**Review by bjornalm — APPROVED** [review]
LGTM

**Pull request merged**